x-pack/plugins/siem/public/components/page/network/domains_table/columns.tsx

```diff
@@ -0,0 +1,198 @@
+          directions.map((direction, index) => (
+            <AddToKql
+              key={escapeDataProviderId(
+                `${tableId}-table-${flowTarget}-${flowDirection}-direction-${direction}`
+              )}
+              expression={`network.direction: ${escapeQueryValue(direction)}`}
+              type={type}
+              componentFilterType={'network'}
+            >
+              <>
+                {defaultToEmptyTag(direction)}
+                {index < directions.length - 1 ? '\u00A0' : null}
+              </>
+            </AddToKql>
+          )),
+  },
+  {
+    field: 'node.network.bytes',
+    name: i18n.BYTES,
+    truncateText: false,
+    hideForMobile: false,
+    sortable: true,
+    render: (bytes: number | null | undefined) => {
+      if (bytes != null) {
+        return numeral(bytes).format('0.000b');
+      } else {
+        return getEmptyTagValue();
+      }
+    },
+  },
+  {
+    field: 'node.network.packets',
+    name: i18n.PACKETS,
+    truncateText: false,
+    hideForMobile: false,
+    sortable: true,
+    render: (packets: number | null | undefined) => {
+      if (packets != null) {
+        return numeral(packets).format('0,000');
+      } else {
+        return getEmptyTagValue();
+      }
+    },
+  },
+  {
+    field: `node.${flowTarget}.uniqueIpCount`,
+    name: getFlowTargetTitle(flowTarget),
+    truncateText: false,
+    hideForMobile: false,
+    sortable: true,
+    render: (uniqueIpCount: number | null | undefined) => {
+      if (uniqueIpCount != null) {
+        return numeral(uniqueIpCount).format('0,000');
+      } else {
+        return getEmptyTagValue();
+      }
+    },
+  },
+  {
+    name: i18n.FIRST_SEEN,
+    truncateText: false,
+    hideForMobile: false,
+    render: ({ node }) => {
+      const domainNameAttr = `${flowTarget}.domainName`;
+      const domainName = getOr(null, domainNameAttr, node);
+      if (domainName != null) {
+        return (
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={flowTarget}
+            type="first-seen"
+          />
+        );
+      }
+      return getEmptyTagValue();
+    },
+  },
+  {
+    name: i18n.LAST_SEEN,
+    truncateText: false,
+    hideForMobile: false,
+    render: ({ node }) => {
+      const domainNameAttr = `${flowTarget}.domainName`;
+      const domainName = getOr(null, domainNameAttr, node);
+      if (domainName != null) {
+        return (
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={flowTarget}
+            type="last-seen"
+          />
+        );
+      }
+      return getEmptyTagValue();
+    },
+  },
+];
+
+const getFlowTargetTitle = (flowTarget: FlowTarget) => {
+  if (flowTarget === FlowTarget.source) {
+    return i18n.UNIQUE_DESTINATIONS;
+  } else if (flowTarget === FlowTarget.destination) {
+    return i18n.UNIQUE_SOURCES;
+  }
+  return '';
```

Comment:
Is this is a legitimate return value for a title, an empty string? Did you want an exhaustive matching here with a `never` or is this correct and you do have the use case of an empty string?

Reply:
No it is not, thanks! The `IP Details` page is currently intended to just support `FlowTarget.source` & `FlowTarget.destination` and thus the global `FlowTargetSelect` is locked down to only show those two values, so the remaining cases (client/server) should never happen.

Thanks for your comment though, as that said, that doesn't mean the `Domains Table` _shouldn't_ support the other two `FlowTarget`'s. :) I went ahead and refactored this as a `switch` with all possible FlowTargets, and `default`ed to a new consolidated `assertUnreachable()` helper I added to `x-pack/plugins/siem/public/lib/helpers/index.tsx `

``` javascript
/** 
 * Unreachable Assertion helper for scenarios like exhaustive switches
 *
 * @param x Unreachable field
 * @param message Message of error thrown
 */
export const assertUnreachable = (
  x: never,
  message = 'Unknown Field in switch statement'
): never => {
  throw new Error(`${message}: ${x}`);
};
```

Turns out there were a few floating around the code base, so this cleans up the duplication a bit -- thanks for pointing this out! :)